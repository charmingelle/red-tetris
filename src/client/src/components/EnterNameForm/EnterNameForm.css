.enter-name-form {
  width: 315px;
  display: flex;
  flex-direction: column;
}

.enter-name-label {
  margin-bottom: 10px;
  font-size: 36px;
  color: #f9f9f9;
}

.enter-name-error {
  margin-bottom: 10px;
  font-size: 30px;
  color: #c00;
}

.enter-name-input {
  height: 45px;
  margin-bottom: 10px;
  border: 5px solid;
  border-left-color: #3033f2;
  border-top-color: #3033f2;
  border-right-color: #020279;
  border-bottom-color: #020279;
  font-size: 36px;
}

.enter-name-input:focus {
  outline: none;
}

.enter-name-button {
  background-color: #0d05f3;
  border: 5px solid;
  border-left-color: #3033f2;
  border-top-color: #3033f2;
  border-right-color: #020279;
  border-bottom-color: #020279;
  color: #fff;
  font-size: 30px;
  font-weight: bold;
  padding: 10px;
}

.enter-name-button:focus {
  outline: none;
}
