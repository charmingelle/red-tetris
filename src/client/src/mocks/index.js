export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
