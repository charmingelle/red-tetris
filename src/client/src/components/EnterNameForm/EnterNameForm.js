import React, { useState } from 'react';
import { connect } from 'react-redux';
import './EnterNameForm.css';
import { isNameValid } from '../../utils';

const saveName = (name, setName, socket) => {
  if (isNameValid(name)) {
    socket.emit('set-name', { name });
    setName('');
  }
};

const handleNameChange = setName => ({ target: { value } }) => setName(value);

const handleKeyPress = (name, setName, socket) => ({ key }) => {
  if (key === 'Enter') {
    saveName(name, setName, socket);
  }
};

const handleButtonClick = (name, setName, socket) => () =>
  saveName(name, setName, socket);

const EnterNameFormInner = ({ socket }) => {
  const [name, setName] = useState('');

  return (
    <div className="enter-name-form">
      <div className="enter-name-label">What's your name?</div>
      <input
        className="enter-name-input"
        value={name}
        type="text"
        placeholder="Enter name..."
        onChange={handleNameChange(setName)}
        onKeyPress={handleKeyPress(name, setName, socket)}
        maxLength="15"
      />
      <button
        className="enter-name-button"
        onClick={handleButtonClick(name, setName, socket)}
      >
        Confirm
      </button>
    </div>
  );
};

const mapStateToPros = ({ socket }) => ({ socket });

export const EnterNameForm = connect(mapStateToPros)(EnterNameFormInner);
