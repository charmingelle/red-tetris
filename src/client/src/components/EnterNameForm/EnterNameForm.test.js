import React from 'react';
import { EnterNameForm } from './EnterNameForm';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { EMPTY_STORE } from '../../mocks';

const mockStore = configureStore([]);

test('EnterNameForm', () => {
  const store = mockStore(EMPTY_STORE);

  expect(
    renderer
      .create(
        <Provider store={store}>
          <EnterNameForm />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
