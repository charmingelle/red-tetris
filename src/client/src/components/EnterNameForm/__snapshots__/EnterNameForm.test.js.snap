// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterNameForm 1`] = `
<div
  className="enter-name-form"
>
  <div
    className="enter-name-label"
  >
    What's your name?
  </div>
  <input
    className="enter-name-input"
    maxLength="15"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter name..."
    type="text"
    value=""
  />
  <button
    className="enter-name-button"
    onClick={[Function]}
  >
    Confirm
  </button>
</div>
`;
